test/Dialect/FIRRTL/expand-whens.mlir

```diff
@@ -405,4 +405,61 @@ firrtl.module @analog(out %analog : !firrtl.analog<1>) {
+  %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+  %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  %0 = firrtl.subindex %ret[0] : !firrtl.vector<uint<1>, 1>
+  firrtl.connect %0, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  firrtl.connect %0, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  // CHECK:      %0 = firrtl.subindex %ret[0] : !firrtl.vector<uint<1>, 1>
+  // CHECK-NEXT: firrtl.connect %0, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1
+}
+
+// CHECK-LABEL: @shadow_when_vector
+firrtl.module @shadow_when_vector(in %p : !firrtl.uint<1>) {
+  %c0_ui2 = firrtl.constant 0 : !firrtl.uint<2>
+  %c1_ui2 = firrtl.constant 1 : !firrtl.uint<2>
+  firrtl.when %p {
+    %w = firrtl.wire : !firrtl.vector<uint<2>, 1>
+    %0 = firrtl.subindex %w[0] : !firrtl.vector<uint<2>, 1>
+    firrtl.connect %0, %c0_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+    firrtl.connect %0, %c1_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+  }
+  // CHECK:      %w = firrtl.wire  : !firrtl.vector<uint<2>, 1>
+  // CHECK-NEXT: %0 = firrtl.subindex %w[0] : !firrtl.vector<uint<2>, 1>
+  // CHECK-NEXT: firrtl.connect %0, %c1_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+}
+
+// CHECK-LABEL: @multi_dim_vector
+firrtl.module @multi_dim_vector(in %p : !firrtl.uint<1>) {
+  %c0_ui2 = firrtl.constant 0 : !firrtl.uint<2>
+  %c1_ui2 = firrtl.constant 1 : !firrtl.uint<2>
+  %w = firrtl.wire : !firrtl.vector<vector<uint<2>, 2>, 1>
+  %0 = firrtl.subindex %w[0] : !firrtl.vector<vector<uint<2>, 2>, 1>
+  %1 = firrtl.subindex %0[0] : !firrtl.vector<uint<2>, 2>
+  %2 = firrtl.subindex %0[1] : !firrtl.vector<uint<2>, 2>
+  firrtl.when %p {
+    firrtl.connect %1, %c0_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+    firrtl.connect %2, %c1_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+  } else {
+    firrtl.connect %1, %c1_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+    firrtl.connect %2, %c0_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+  }
+  // CHECK:      [[MUX1:%.*]] = firrtl.mux(%p, %c0_ui2, %c1_ui2) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+  // CHECK-NEXT: firrtl.connect %1, [[MUX1]] : !firrtl.uint<2>, !firrtl.uint<2>
+  // CHECK-NEXT: [[MUX2:%.*]] = firrtl.mux(%p, %c1_ui2, %c0_ui2) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+  // CHECK-NEXT: firrtl.connect %2, [[MUX2]] : !firrtl.uint<2>, !firrtl.uint<2>
+}
+
+// CHECK-LABEL: @vector_of_bundle
+firrtl.module @vector_of_bundle(in %p : !firrtl.uint<1>, out %ret: !firrtl.vector<bundle<a:uint<1>>, 1>) {
+  %0 = firrtl.subindex %ret[0] : !firrtl.vector<bundle<a:uint<1>>, 1>
+  %1 = firrtl.subfield %0(0) : (!firrtl.bundle<a:uint<1>>) -> !firrtl.uint<1>
+  %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+  %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  firrtl.connect %1, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  firrtl.connect %1, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  // CHECK: firrtl.connect %1, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
```

Comment:
You could add a `CHECK-NOT` to these to make sure that this is removing the first connection. I think that would have to come _before_ the `CHECK` to make sure that it wasn't using the earlier one.